`TestHSMDualAuthRotation` flakiness

## Failure

#### Link(s) to logs

- https://console.cloud.google.com/cloud-build/builds/bf5dc298-970e-468c-b739-11b47347036f?project=ci-account
- https://console.cloud.google.com/cloud-build/builds/d44b12f3-f37a-4835-a443-e3b5db10c712?project=ci-account
- https://console.cloud.google.com/cloud-build/builds/143a9e10-c336-4dc4-baa5-3846cd40d7d3?project=ci-account
- https://console.cloud.google.com/cloud-build/builds/1e717527-9d6e-482c-bce1-13182af4468d?project=ci-account
- 07.07.2022 -> https://console.cloud.google.com/cloud-build/builds;region=us-west1/e42873ae-6da5-4f76-a6f7-687771a0997f;step=0?project=ci-account
#### Relevant snippet

```
    hsm_test.go:76: 
        	Error Trace:	hsm_test.go:76
        	            				testing.go:1034
        	            				testing.go:1203
        	            				testing.go:1187
        	            				panic.go:500
        	            				testing.go:864
        	            				hsm_test.go:76
        	            				testing.go:1034
        	            				testing.go:1203
        	            				testing.go:1433
        	            				panic.go:500
        	            				testing.go:864
        	            				hsm_test.go:680
        	            				hsm_test.go:693
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestHSMDualAuthRotation
        	Messages:   	error while closing auth1 during test cleanup
--- FAIL: TestHSMDualAuthRotation (480.13s)
```

--- a/integration/hsm/hsm_test.go
+++ b/integration/hsm/hsm_test.go
@@ -341,7 +341,6 @@ func liteBackendConfig(t *testing.T) *backend.Config {
 
 // Tests a single CA rotation with a single HSM auth server
 func TestHSMRotation(t *testing.T) {
-	t.Parallel()
 	if os.Getenv("SOFTHSM2_PATH") == "" {
 		t.Skip("Skipping test as SOFTHSM2_PATH is not set")
 	}
@@ -407,7 +406,6 @@ func TestHSMRotation(t *testing.T) {
 
 // Tests multiple CA rotations and rollbacks with 2 HSM auth servers in an HA configuration
 func TestHSMDualAuthRotation(t *testing.T) {
-	t.Parallel()
 	if os.Getenv("TELEPORT_ETCD_TEST") == "" || os.Getenv("SOFTHSM2_PATH") == "" {
 		t.Skip("Skipping test as either etcd or SoftHSM2 is not enabled")
 	}
@@ -696,7 +694,6 @@ func TestHSMDualAuthRotation(t *testing.T) {
 
 // Tests a dual-auth server migration from raw keys to HSM keys
 func TestHSMMigrate(t *testing.T) {
-	t.Parallel()
 	if os.Getenv("TELEPORT_ETCD_TEST") == "" || os.Getenv("SOFTHSM2_PATH") == "" {
 		t.Skip("Skipping test as either etcd or SoftHSM2 is not enabled")
 	}

